frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -0,0 +1,330 @@
+      cells: [],
+      selected: false,
+      id: node.metadata.name,
+      metadata: _.clone(node.metadata),
+      spec: _.clone(node.spec),
+    };
+    obj.cells = [
+      {
+        title: <ResourceLink kind="Node" name={node.metadata.name} title={node.metadata.uid} />,
+      },
+      {
+        title: roles.join(', ') || '-',
+      },
+      {
+        title: `${humanizeCpuCores(_.get(node.status, 'capacity.cpu')).string || '-'} CPU`,
+      },
+      {
+        title: `${getConvertedUnits(_.get(node.status, 'allocatable.memory'), 'KiB')}`,
+      },
+    ];
+    return obj;
+  });
+};
+
+const getAllTableFilters = () => ({
+  ...tableFilters,
+});
+
+const getFilteredRows = (_filters, objects) => {
+  if (_.isEmpty(_filters)) {
+    return objects;
+  }
+
+  const allTableFilters = getAllTableFilters();
+  let filteredObjects = objects;
+  _.each(_filters, (value, name) => {
+    const filter = allTableFilters[name];
+    if (_.isFunction(filter)) {
+      filteredObjects = _.filter(filteredObjects, (o) => filter(value, o));
+    }
+  });
+
+  return filteredObjects;
+};
+
+const getPreSelectedNodes = (nodes) => {
+  nodes.forEach((node) => {
+    node.selected = _.get(node, `metadata.labels['${ocsLabel}']`) !== undefined;
+  });
+};
+
+const stateToProps = (obj, { data = [], filters = {}, staticFilters = [{}] }) => {
+  const allFilters = staticFilters ? Object.assign({}, filters, ...staticFilters) : filters;
+  const newData = getFilteredRows(allFilters, data);
+  return {
+    data: newData,
+    unfilteredData: data,
+  };
+};
+
+const CustomNodeTable: React.FC<CustomNodeTableProps> = ({ data, loaded, ocsProps }) => {
+  const columns = getColumns();
+  const [nodes, setNodes] = React.useState([]);
+  const [sortBy, setSortBy] = React.useState<ISortBy>({ index: 0, direction: 'asc' });
+  const [error, setError] = React.useState('');
+  const [inProgress, setProgress] = React.useState(false);
+  const [storageClass, setStorageClass] = React.useState('');
+  const [selectedNodesCnt, setSelectedNodesCnt] = React.useState(0);
+
+  React.useEffect(() => {
+    const selectedNode = _.filter(nodes, 'selected').length;
+    setSelectedNodesCnt(selectedNode);
+  }, [nodes]);
+
+  React.useEffect(() => {
+    const formattedNodes = getRows(data);
+    // pre-selection of nodes
+    if (loaded && !nodes.length) {
+      getPreSelectedNodes(formattedNodes);
+      setNodes(formattedNodes);
+    }
+    // for getting nodes
+    else if (formattedNodes.length) {
+      nodes.forEach((row) => {
+        if (row.selected) {
+          const index = formattedNodes.findIndex((r) => r.id === row.id);
+          formattedNodes[index].selected = true;
+        }
+      });
+      setNodes(formattedNodes);
+    }
+  }, [data, loaded, nodes]);
+
+  // sorting by name
+  const onSort = (event, index, direction) => {
+    event.preventDefault();
+    const sortedNodes = nodes.sort((n1, n2) => (n1.id < n2.id ? -1 : n1.id > n2.id ? 1 : 0));
```

Comment:
This looks wrong... Isn't this always sorting by `id` no matter what `index` is?

Consider `_.orderBy`